src/lib/util.ts

```diff
@@ -391,6 +393,57 @@ export function hasGpRewards(reward: IRewardState) {
   return false;
 }
 
+/**
+ * Returns an object describing non-zero, unredeemed Contribution reward amounts for the current dao
+ * @param proposals list of proposal with unclaimed rewards
+ */
+export function getDaoDebt(proposals: IProposalState[]): AccountClaimableRewardsType {
+  const result: AccountClaimableRewardsType = {};
+
+  let ethReward;
+  let externalTokenReward;
+  let rewards;
+
+  const supportedExternalTokens = supportedTokens();
+
+  result["ETH"] = new BN(0);
+  for (const proposal of proposals) {
+
+    // Calculate CR Debt
+    if (proposal.contributionReward && proposal.contributionReward.externalToken !== utils.NULL_ADDRESS ) {
```

Comment:
Is the second part of this if statement needed? What if a CR proposal doesn't have an externalToken as a requested reward? For example, if it's only asking for ETH, this if statement would fail no?

Reply:
Nice eye! The second part of if statement was supposed to be separate check in token debt section and I messed up. Thank you

Reply:
Nice Eye!

I messed up the loging while trying to exclude the proposals which cannot be redeemed because of null externalToken but non zero ethReward and externalTokenReward.